esp/templates/email/class_cancellation_body.txt

```diff
@@ -1,4 +1,5 @@
-We regret to inform you that class section {{ sec.emailcode }}: {{ sec.title }} at the upcoming {{ prog.niceName }} has been cancelled.  {% if sec.friendly_times|length_is:1 %}The class had been scheduled for {{ sec.friendly_times.0 }}, so you can switch into another class during that period.  You can do this on our Web site or on-site at {{ prog.program_type }}.{% else %}You can sign up for other classes on our Web site or on-site at {{ prog.program_type }}.{% endif %}
+We regret to inform you that class {% templatetag openblock %} if classreg {% templatetag closeblock %}section {{ sec.emailcode }}{% templatetag openblock %} else {% templatetag closeblock %}{{ sec.parent_class.emailcode }}{% templatetag openblock %} endif {% templatetag closeblock %}: {{ sec.title }} at the upcoming {{ prog.niceName }} has been cancelled.  {% templatetag openblock %} if classreg {% templatetag closeblock %}{% if sec.friendly_times|length_is:1 %}The class had been scheduled for {{ sec.friendly_times.0 }}, so you can switch into another class during that period.  You can do this on our Web site or on-site at {{ prog.program_type }}.{% else %}You can sign up for other classes on our Web site or on-site at {{ prog.program_type }}.{% endif %}{% templatetag openblock %} else {% templatetag closeblock %}You can sign up for other classes on our Web site or on-site at {{ prog.program_type }}.{% templatetag openblock %} endif {% templatetag closeblock %}
```

Comment:
I'm not super-happy with the duplication of the else-block here, but I'm not sure that there's a good way to fix it with the current double-render strategy.  Also, I think admins especially at places that aren't MIT may be confused by seeing template tags in the email they get.  Since the only thing that changes is `classreg` it might be worth just having the admin email include both bodies successively if we emailed students with SSIs.  This should allow removing all the `{% templatetag openblock %}` stuff and simplify the conditionals in this template a bit.